server/app/assets/javascripts/admin_predicate_configuration.ts

```diff
@@ -0,0 +1,356 @@
+      switch (selectedScalarType.toUpperCase()) {
+        case 'STRING':
+          if (selectedScalarValue.toUpperCase() === 'EMAIL') {
+            // Need to look at the selected scalar *value* for email since the type is just a
+            // string, but emails have a special type in HTML inputs.
+            valueInput.setAttribute('type', 'email')
+            break
+          }
+          valueInput.setAttribute('type', 'text')
+          break
+        case 'CURRENCY_CENTS':
+          if (
+            operatorValue.toUpperCase() === 'IN' ||
+            operatorValue.toUpperCase() === 'NOT_IN'
+          ) {
+            // IN and NOT_IN operate on lists of longs, which must be entered as a comma-separated list
+            valueInput.setAttribute('type', 'text')
+          } else {
+            valueInput.setAttribute('step', '.01')
+            valueInput.setAttribute('placeholder', '$0.00')
+            valueInput.setAttribute('type', 'number')
+          }
+          break
+        case 'LONG':
+          if (
+            operatorValue.toUpperCase() === 'IN' ||
+            operatorValue.toUpperCase() === 'NOT_IN'
+          ) {
+            // IN and NOT_IN operate on lists of longs, which must be entered as a comma-separated list
+            valueInput.setAttribute('type', 'text')
+          } else {
+            valueInput.setAttribute('step', '1')
+            valueInput.setAttribute('placeholder', '0')
+            valueInput.setAttribute('type', 'number')
+          }
+          break
+        case 'DATE':
+          valueInput.setAttribute('type', 'date')
+          break
+        default:
+          valueInput.setAttribute('type', 'text')
+      }
+    }
+  }
+
+  /** Add a value row by duplicating the last row in the list and updating its group IDs. */
+  predicateAddValueRow(event: Event) {
+    event.preventDefault()
+    event.stopPropagation()
+
+    const currentRows = document.querySelectorAll('.predicate-config-value-row')
+    const templateRow = currentRows[currentRows.length - 1]
+    const newRow = templateRow.cloneNode(true) as HTMLElement
+
+    newRow.querySelectorAll('input').forEach((el: HTMLInputElement) => {
+      if (el.type === 'checkbox' || el.type === 'radio') {
+        el.checked = false
+      } else {
+        el.value = ''
+      }
+
+      const groupNumString = assertNotNull(el.name.match(/group-(\d+)/))[1]
+
+      let groupNum = parseInt(groupNumString, 10)
+      el.name = el.name.replace(/group-\d+/, `group-${++groupNum}`)
+      const newId = `${el.id}-${groupNum}`
+      el.id = newId
+
+      el.closest('label')?.setAttribute('for', newId)
+    })
+
+    const deleteButtonDiv = assertNotNull(
+      newRow.querySelector('.predicate-config-delete-value-row'),
+    ) as HTMLElement
+
+    deleteButtonDiv.addEventListener('click', (event: Event) =>
+      this.predicateDeleteValueRow(event),
+    )
+    assertNotNull(deleteButtonDiv.querySelector('svg')).classList.remove(
```

Comment:
Curious, why would the button be hidden?

Guessing, but does the first row have a permanent hidden button and this is accounting for the first copy?  Eitherway comment it.

Reply:
done